File: go/url.go
package main

import (
	"fmt"
	"log"
	"net/url"
)

func decode(encodedValue string) {
	fmt.Println("")
	fmt.Println("encodedValue=", encodedValue)
	decodedValue, err := url.QueryUnescape(encodedValue)
	if err != nil {
		log.Fatal(err)
		return
	}
	fmt.Println("decodedValue=", decodedValue)
}

func main() {
	decode("Hello World")
	decode("Hello+World")
	decode("Hello%20World")
}

File: php/url.php
<?php

function decode($str) {
  print "-------\n";
  print $str . "\n";
  print urldecode($str) . "\n";
}

decode('a+b');
decode('a b');
decode('a%20b');
$str = "a%20b";
print $str . "\n";
?>
